src/typeahead/highlight.ts

```diff
@@ -30,28 +30,30 @@ export class NgbHighlight implements OnChanges {
    * The text highlighting is added to.
    *
    * If the `term` is found inside this text, it will be highlighted.
+   * If the `term` contains array then all the items from it will be highlighted inside the text.
    */
-  @Input() result: string;
+  @Input() result: any;
```

Comment:
Don't see the necessity of changing this at the moment. 

The issue is about supporting multiple terms, also this is not covered by tests

Reply:
`any` probably is too much freedom. I see we have test coverage for string and number. Is it ok to change the type to `string | number`?